src/RatingComponent.js

```diff
@@ -0,0 +1,270 @@
+import React, { Component } from 'react'
+import {
+  StyleSheet,
+  Text,
+  View,
+  TouchableOpacity,
+  Modal,
+  TextInput,
+  Linking,
+  Platform,
+  KeyboardAvoidingView
+} from 'react-native'
+import PropTypes from 'prop-types'
+import getRatingType, { setAlpha, setShowDate } from './utils'
+import RatingsDisplay from './RatingsDisplay';
+
+const KeyBoardAvoidView = Platform.OS === 'ios' ? KeyboardAvoidingView : View
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    flexDirection: 'column',
+    width: '100%',
+    alignItems: 'center',
+    marginBottom: 0,
+    marginTop: 0,
+    justifyContent: 'flex-end'
+  },
+  thanksText: {
+    fontSize: 14,
+    fontFamily: 'Arial',
+    color: '#000000',
+    textAlign: 'center'
+  },
+  thanksView: {
+    height: 120,
+    width: 265,
+    backgroundColor: '#ffffff',
+    justifyContent: 'center',
+    alignItems: 'center'
+  },
+  title: {
+    fontSize: 16,
+    textAlign: 'center',
+    fontFamily: 'Arial',
+    flexWrap: 'wrap',
+    color: '#292929',
+    marginTop: 24
+  },
+  subtext: {
+    fontSize: 12,
+    marginTop: 4,
+    textAlign: 'center',
+    fontFamily: 'Arial',
+    color: '#292929'
+  },
+  later: {
+    textAlign: 'center',
+    fontSize: 14,
+    color: '#292929',
+    marginBottom: 25,
+    opacity: 0.5
+  },
+  button: {
+    height: 36,
+    width: 328,
+    backgroundColor: '#1fd290',
+    borderRadius: 4,
+    alignSelf: 'center',
+    justifyContent: 'center',
+    alignItems: 'center'
+  },
+  buttonText: {
+    color: '#ffffff',
+    fontSize: 14,
+    fontFamily: 'Arial',
+    textAlign: 'center'
+  },
+  input: {
+    width: '100%'-  32,
+    marginLeft: 16,
+    marginRight: 16
+  }
+})
+
+export default class RatingComponent extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      rateVisible: true,
+      thanksVisible: false,
+      showButton: false,
+      showInputText: false,
+      rating: 1,
+      feedback: ''
+    }
+  }
+
+  componentDidMount(){
+    this.props.sendEvent({ type: 'ratings' })
```

Comment:
make 'rating' string type to a meaningful string like 'RATING_MOUNTED' or 'RATING OPENED'